lib/surface/components/form/field_context.ex

```diff
@@ -12,7 +12,7 @@ defmodule Surface.Components.Form.FieldContext do
   alias Surface.Components.Form.Field
 
   @doc "The field name"
-  prop name, :atom, required: true
+  prop name, :any, required: true
```

Comment:
@miguel-s I believe this change makes it consistent with the changes in https://github.com/surface-ui/surface/pull/319/. Correct? Is there anything else to be done?

Reply:
Looks good to me 👍 

We might want to add a test to `field_contest_test.exs` to check `name` with atom, we only have one for strings now.